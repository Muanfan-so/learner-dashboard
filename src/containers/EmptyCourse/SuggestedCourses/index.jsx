import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Card, Button, Container } from '@edx/paragon';
import { hooks as appHooks } from 'data/redux';

import messages from './messages';

export const SuggestedCourses = () => {
  const suggestedCourses = appHooks.useSuggestedCoursesData();
  const { formatMessage } = useIntl();
  if (!suggestedCourses.length) { return null; }
  return (
    <Container size="md">
      <h3 className="text-center">{formatMessage(messages.header)}</h3>
      <div className="d-flex">
        {suggestedCourses.map((course) => (
          <Card key={course.title} className="m-3">
            <Card.ImageCap
              src={course.bannerUrl}
              srcAlt={formatMessage(messages.courseImageAlt)}
              logoSrc={course.logoUrl}
              logoAlt={formatMessage(messages.institueLogoAlt)}
            />
            <Card.Header title={course.title} />
            <Card.Footer>
              <Button as="a">{formatMessage(messages.viewCourseButton)}</Button>
            </Card.Footer>
          </Card>
        ))}
      </div>
    </Container>
  );
};

export default SuggestedCourses;
