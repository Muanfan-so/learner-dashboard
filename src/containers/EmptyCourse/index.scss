@import "@edx/paragon/scss/core/core";

.empty-course-hero {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: map-get($spacers, 5);

  & > * {
    margin-top: map-get($spacers, 3);
    margin-bottom: map-get($spacers, 3);
  }
}