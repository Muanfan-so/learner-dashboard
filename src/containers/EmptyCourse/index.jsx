import React from 'react';
import { FormattedMessage } from '@edx/frontend-platform/i18n';
import { Button } from '@edx/paragon';

import SuggestedCourses from './SuggestedCourses';
import messages from './messages';
import './index.scss';

export const EmptyCourse = () => (
  <div className="p-3">
    <div className="empty-course-hero">
      <img src="empty-course.svg" alt="empty course banner" />
      <h1>
        <FormattedMessage {...messages.lookingForChallengePrompt} />
      </h1>
      <p>
        <FormattedMessage {...messages.exploreCoursesPrompt} />
      </p>
      <Button variant="brand">
        <FormattedMessage {...messages.exploreCoursesButton} />
      </Button>
    </div>
    <SuggestedCourses />
  </div>
);

export default EmptyCourse;
