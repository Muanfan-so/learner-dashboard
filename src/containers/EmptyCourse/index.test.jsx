import React from 'react';
import { shallow } from 'enzyme';

import EmptyCourse from '.';

jest.mock('./SuggestedCourses', () => 'SuggestedCourses');

describe('EmptyCourse', () => {
  test('snapshot', () => {
    expect(shallow(<EmptyCourse />)).toMatchSnapshot();
  });
});
