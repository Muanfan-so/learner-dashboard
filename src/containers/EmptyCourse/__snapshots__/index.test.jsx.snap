// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCourse snapshot 1`] = `
<div
  className="p-3"
>
  <div
    className="empty-course-hero"
  >
    <img
      alt="empty course banner"
      src="empty-course.svg"
    />
    <h1>
      <FormattedMessage
        defaultMessage="Looking for a new challenge?"
        description="Prompt user for new challenge"
        id="EmptyCourse.lookingForChallengePrompt"
      />
    </h1>
    <p>
      <FormattedMessage
        defaultMessage="Explore our courses to add them to your dashboard."
        description="Prompt user to explore more courses"
        id="EmptyCourse.exploreCoursesPrompt"
      />
    </p>
    <Button
      variant="brand"
    >
      <FormattedMessage
        defaultMessage="Explore courses"
        description="Button to explore more courses"
        id="EmptyCourse.exploreCoursesButton"
      />
    </Button>
  </div>
  <SuggestedCourses />
</div>
`;
